Feat/AACT-246 create model release

Javier-Jimenez-18 opened the pull request:
Added flatpickr css style link to app layout. Fixed EventListener so that flatpickr works properly. Updated Releases index page so that Release Title links function and Releases Delete buttons function. Added a Releases Show page.

<img width="1280" alt="Screen Shot 2021-12-09 at 4 18 25 PM" src="https://user-images.githubusercontent.com/81119399/145498429-9cddce43-a4b2-4d83-b7a6-fa83be42e08c.png">

<img width="1280" alt="Screen Shot 2021-12-09 at 4 21 33 PM" src="https://user-images.githubusercontent.com/81119399/145498435-b9832945-69cc-4858-b096-a7dd63ef30b9.png">

<img width="1280" alt="Screen Shot 2021-12-09 at 4 22 55 PM" src="https://user-images.githubusercontent.com/81119399/145498437-b67bafea-2180-477c-a58a-484b962f18de.png">

<img width="1280" alt="Screen Shot 2021-12-09 at 4 23 36 PM" src="https://user-images.githubusercontent.com/81119399/145498439-e9b68263-42be-4faa-9b1a-db16d982521c.png">

<img width="1280" alt="Screen Shot 2021-12-09 at 4 25 42 PM" src="https://user-images.githubusercontent.com/81119399/145498440-1626fb2f-187a-4527-a689-0c54e9da6693.png">

Commit:
Created model Release

Commit:
Created Releases controller and updated Release model

Commit:
Created file _form.html.erb for Releases

Commit:
Modified Release Notes and Release controller files. Deleted unnecessary view files. Modified Routes and Factories files.

Commit:
Set up Release view files for Index, New, and Edit.

Commit:
Added flatpickr gem, added strftime Ruby date formatting to Release Notes and Releases index pages.

Commit:
Updated Release Notes and Releases controller files so that the releases are ordered by the released_on date.

Commit:
Added flatpickr css style link to app layout. Fixed EventListener so that flatpickr works properly. Updated Releases index page so that Releases Titles link functions and Releases Delete buttons function. Added a Releases Show page.

Commit:
merged and resolved the conflict in Gemfile.lock, structure.sql, and factories.rb

Commit:
resolved branch conflicts.

Commit:
ran mgrations.

crystal-wb reviewed (changes requested):
Looks pretty good. There is one thing that needs fixing in the structure file. And I had some suggestions for improvements.

Commit:
Added Edit button on releases show page. Deleted unnecessary files and comments. Modidied releases and release notes controllers to order them using released_on and not created_at.

Javier-Jimenez-18 commented:
Added Edit button on releases show page. Deleted unnecessary files and comments. Modified releases and release notes controllers to order them using released_on and not created_at."

<img width="1280" alt="Screen Shot 2021-12-21 at 3 57 31 PM" src="https://user-images.githubusercontent.com/81119399/147012637-45e1e785-60c7-4db8-9eae-b3952f62dac4.png">

bykasya reviewed (changes requested):
Looks great, @Javier-Jimenez-18! 👍🏻  I just requested a few minor changes, and also to resolve conflict with dev.

Javier-Jimenez-18 committed:
Merge branch 'dev' into feat/AACT-246-create_model_release

Commit:
Removed json format from detroy method in releases controller file.

crystal-wb reviewed (changes requested):
Looks good.

Could you add some tests for the controller? Response tests would be good. I don't think we need a lot of tests but it would be good to test the controller actions/response.

Javier-Jimenez-18 committed:
merged dev

Javier-Jimenez-18 committed:
Removed format for detroy method

Javier-Jimenez-18 committed:
Updated Gemfile and ran migrations. Updated scss stylesheets for table and button styling. Updated Release pages and form for button and table styling.

Javier-Jimenez-18 commented:
Fixed the button and table styling for Releases.

Releases index:
<img width="1280" alt="Screen Shot 2022-01-25 at 2 06 24 PM" src="https://user-images.githubusercontent.com/81119399/151072236-bd5426fe-abf9-47db-8990-a18c7db2b5e1.png">

Releases new:
<img width="1280" alt="Screen Shot 2022-01-25 at 2 17 37 PM" src="https://user-images.githubusercontent.com/81119399/151072317-ab4f8d77-b1d4-4d93-aae2-1253bd2e7534.png">

Releases edit:
<img width="1280" alt="Screen Shot 2022-01-25 at 2 02 24 PM" src="https://user-images.githubusercontent.com/81119399/151072407-1a7501f3-9411-4474-977b-96887874bac1.png">

Releases show:
<img width="1280" alt="Screen Shot 2022-01-25 at 2 02 42 PM" src="https://user-images.githubusercontent.com/81119399/151072509-903e7bab-5eb2-442f-9840-8799302e131d.png">

Releases index (with new Release X created):
<img width="1280" alt="Screen Shot 2022-01-25 at 2 19 25 PM" src="https://user-images.githubusercontent.com/81119399/151072753-37719477-eac6-43f0-92a2-c44749c64c36.png">

Releases delete (deleted Release X):
<img width="1280" alt="Screen Shot 2022-01-25 at 2 20 17 PM" src="https://user-images.githubusercontent.com/81119399/151072901-b3e94a3f-f969-4be1-9e55-34a51008da16.png">

Javier-Jimenez-18 commented:
Completed Request Tests for the Releases.

<img width="1280" alt="Screen Shot 2022-01-25 at 2 24 38 PM" src="https://user-images.githubusercontent.com/81119399/151073207-ea280fe4-1978-4b32-9db1-116def8b0980.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 25 53 PM" src="https://user-images.githubusercontent.com/81119399/151073243-f4803f0b-4280-4f90-8995-6fbc88bd7454.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 27 20 PM" src="https://user-images.githubusercontent.com/81119399/151073266-1fc2a292-7a0b-4c30-8b4b-4fd9e99d14c2.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 28 22 PM" src="https://user-images.githubusercontent.com/81119399/151073289-8a785f06-ca09-4300-8b79-8d40acade072.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 32 33 PM" src="https://user-images.githubusercontent.com/81119399/151073341-285da430-b894-488d-a892-339f2d1093cf.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 33 38 PM" src="https://user-images.githubusercontent.com/81119399/151073388-71986fef-4a0b-48a7-b9b6-baecbbbdef0b.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 35 09 PM" src="https://user-images.githubusercontent.com/81119399/151073424-1b8deda5-cde0-4f99-ba3b-6a75ebf0384f.png">

<img width="1280" alt="Screen Shot 2022-01-25 at 2 35 38 PM" src="https://user-images.githubusercontent.com/81119399/151073466-ad61abed-fb32-4016-8ab3-a4933fed3839.png">


Rspec Tests:

<img width="1280" alt="Screen Shot 2022-01-25 at 2 38 18 PM" src="https://user-images.githubusercontent.com/81119399/151073572-70250de8-efa0-4a94-b291-80463757fea3.png">

Javier-Jimenez-18 committed:
Added Requests tests for the Releases.

Javier-Jimenez-18 committed:
Updated is_admin? method in Application Controller. Added is_admin? call in Releases Controller. Added Devise sign-in method test helpers in SPEC Rails Helper. Added Releases Requests tests for Users as an admin, non-admin, and not logged in.
